OpenRA.Mods.Common/Traits/Player/ClassicProductionQueue.cs

```diff
@@ -104,7 +104,7 @@ protected override bool BuildUnit(ActorInfo unit)
 				return false;
 			}
 
-			foreach (var p in producers.Where(p => !p.Actor.IsDisabled()))
+			foreach (var p in producers.Where(p => !p.Trait.IsTraitPaused && !p.Trait.IsTraitDisabled))
```

Comment:
I think it would fit better to put the `IsTraitDisabled` check as part of the `producers` lookup above.  If all production traits are disabled (as opposed to paused) then we should cancel the production and give a refund.